flutter/third_party/android/android_configure.bzl

```diff
@@ -0,0 +1,62 @@
+"""Repository rule for Android SDK and NDK autoconfiguration.
+
+This rule is a no-op unless the required android environment variables are set.
+"""
+
+# Based on https://github.com/tensorflow/tensorflow/tree/34c03ed67692eb76cb3399cebca50ea8bcde064c/third_party/android
+# Workaround for https://github.com/bazelbuild/bazel/issues/14260
+
+_ANDROID_NDK_HOME = "ANDROID_NDK_HOME"
+_ANDROID_SDK_HOME = "ANDROID_HOME"
+
+def _escape_for_windows(path):
+    """Properly escape backslashes for Windows.
+
+    Ideally, we would do this conditionally, but there is seemingly no way to
+    determine whether or not this is being called from Windows.
+    """
+    return path.replace("\\", "\\\\")
+
+def _android_autoconf_impl(repository_ctx):
+    sdk_home = repository_ctx.os.environ.get(_ANDROID_SDK_HOME)
+    ndk_home = repository_ctx.os.environ.get(_ANDROID_NDK_HOME)
+
+    # version 31.0.0 won't work https://stackoverflow.com/a/68036845
+    sdk_rule = ""
+    if sdk_home:
+        sdk_rule = """
+    native.android_sdk_repository(
+        name="androidsdk",
+        path="{}",
+        build_tools_version="30.0.3",
+    )
+""".format(_escape_for_windows(sdk_home))
```

Comment:
sounds right. will do it.

Reply:
@d-uzlov are you sure that on windows this doesn't work? I added this to get around the Android SDK problem in bazel you mentioned on macOS. I met the Android SDK missing problem when I tried to build this on macOS. It should work on Windows also. I mean if `$ANDROID_HOME` is not,  `sdk_home` should not be set.

Reply:
There are no issues building supported targets on Windows.
I don't have android-related environment variables defined on Windows, and bazel doesn't try to load them.

The issue is in the WSL virtual machine.
`ANDROID_HOME` is defined there because some of the Android tools are installed, also NDK is installed there, but the there is no actual SDK because it's not needed.

Reply:
argh, got it, thanks. let me check how to handle this on Linux

Reply:
@d-uzlov Could elaborate more on why we need `ANDROID_HOME`. I was able to build and run flutter app for Android devices on Ubuntu without `ANDROID_HOME` set.

Reply:
We don't need it to build native libraries. And Android SDK is not required.
But if `ANDROID_HOME` is defined for whatever reason then Android SDK becomes required, even though it's not used by bazel.

I don't think requiring users not to have `ANDROID_HOME` is reasonable.

Reply:
Let's merge it and I remove unused dependencies in a separate PR.